### PR TITLE
[#1024] improvement(tez): Optimize user switch to shuffle mode local/remote.

### PR DESCRIPTION
**Pull request by lifeSo**
### What changes were proposed in this pull request?

Currently, users need to set the parameter tez.am.launch.launch.cmd-opts to switch between local shuffle and remote shuffle, but this parameter is difficult for users to understand.
We need a simpler and easier to understand parameter，such as tez.shuffle.mode = local/remote
### Why are the changes needed?

> [Improvement] [tez]：Optimize user switch to shuffle mode local/remote. #1024

Fix:  https://github.com/apache/incubator-uniffle/issues/1024

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

unit test or cluster prd test

### CODE REVIEW

**Comment by lifeSo** [comment]
@baitian77  
is this PR satifify your requirement ?
Task use RSS default, 
and we can set "tez.shuffle.mode=false" to fail back to local mode.

**Comment by jerqi** [comment]
@zhengchenyu Could you help me review this pr?

File: client-tez/src/main/java/org/apache/tez/dag/app/RssDAGAppMaster.java [hunk 1]
```diff
@@ -342,6 +342,20 @@ public static void main(String[] args) {
       // reconfigure.
       reconfigureLog4j();
 
+      Configuration conf = new Configuration(new YarnConfiguration());
```
Comment (zhengchenyu) [inline client-tez/src/main/java/org/apache/tez/dag/app/RssDAGAppMaster.java:345]: Can we move this loop to Line 329-339?

Comment (lifeSo) [reply client-tez/src/main/java/org/apache/tez/dag/app/RssDAGAppMaster.java:345]: No, the code before is init log class and log config.
image

Comment (lifeSo) [reply client-tez/src/main/java/org/apache/tez/dag/app/RssDAGAppMaster.java:345]: If move the code to  Line 329, then will be no am task log due to no log class and "log4j.configuration" config.

eg:
img_v3_026o_3f299d75-7575-4406-a473-9bf24f3dbafg

**Review by zhengchenyu — APPROVED** [review]

File: docs/client_guide/tez_client_guide.md [hunk 2]
```diff
@@ -45,6 +45,7 @@ Note that the RssDAGAppMaster will automatically disable slow start (i.e., `tez.
 | tez.rss.client.max.buffer.size | 3k | The max buffer size in map side. Control the size of each segment(WrappedBuffer) in the buffer. |
 | tez.rss.client.batch.trigger.num | 50 | The max batch of buffers to send data in map side. Affect the number of blocks sent to the server in each batch, and may affect rss_worker_used_buffer_size |
 | tez.rss.client.send.thread.num | 5 | The thread pool size for the client to send data to the server. |
+| tez.shuffle.mode               | remote  | Use Remote Shuffle if the value is set to 'remote' or use default config value, or set 'local' to use local shuffle when needs fall boak.                   |
```
Comment (jerqi) [inline docs/client_guide/tez_client_guide.md:48]: `needs  fall boak` -> `needs to fall back`

**Review by jerqi — APPROVED** [review]
LGTM. merged to master. thanks @lifeSo @zhengchenyu

**Pull request merged**